modules/version-checker/version_checker.go

```diff
@@ -0,0 +1,174 @@
+package version_checker
+
+import (
+	"fmt"
+	"github.com/gruntwork-io/terratest/modules/shell"
+	"github.com/gruntwork-io/terratest/modules/testing"
+	"github.com/hashicorp/go-version"
+	"github.com/stretchr/testify/require"
+	"regexp"
+)
+
+// VersionMismatchErr is an error to indicate version mismatch.
+type VersionMismatchErr struct {
+	expectedVersion string
+	actualVersion   string
+}
+
+func (r *VersionMismatchErr) Error() string {
+	return fmt.Sprintf("found version mismatch {%s} when "+
+		"expecting min version of {%s}", r.expectedVersion, r.actualVersion)
+}
```

Comment:
NIT: put this in a separate file called `errors.go` so it's easier to find all the custom error types.